File: backend/src/main/java/com/atsensei/academy/modules/program/ProgramService.java
package com.atsensei.academy.modules.program;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

@Service
@Transactional(readOnly = true)
public class ProgramService {

    private final ProgramRepository programRepository;

    public ProgramService(ProgramRepository programRepository) {
        this.programRepository = programRepository;
    }

    public List<ProgramEntity> getAllPrograms() {
        return programRepository.findAll();
    }

    public List<ProgramEntity> getProgramsByCategory(String category) {
        return programRepository.findByCategory(category);
    }

    public Optional<ProgramEntity> getProgramByCode(String code) {
        return programRepository.findByCode(code);
    }
}

File: frontend/src/api/programService.js
import { request } from './apiClient';

export const ProgramService = {
  async getAllPrograms(category = '') {
    const query = category ? `?category=${encodeURIComponent(category)}` : '';
    const res = await request(`/programs${query}`);
    return res.data;
  },

  async getProgramByCode(code) {
    const res = await request(`/programs/${code}`);
    return res.data;
  }
};
